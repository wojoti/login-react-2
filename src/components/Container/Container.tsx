import React, { ReactNode } from "react";
import { Wrapper } from "./Container.style";

type Props = {
  children: ReactNode;
};

const Container = (props: Props) => {
  return <Wrapper>{props.children}</Wrapper>;
};

export default Container;
