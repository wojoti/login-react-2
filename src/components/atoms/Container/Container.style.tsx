import styled from "styled-components";

const Wrapper = styled.div`
  box-sizing: border-box;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  display: flex;
  flex-direction: column;
  box-shadow: 0 25px 50px -12px rgb(0 0 0 / 0.25);
  padding-top: 2rem;
  padding-bottom: 2rem;
  padding-left: 2.5rem;
  padding-right: 2.5rem;
  background-color: rgb(255 255 255);
  border-style: solid;
  border-width: 0rem;
  border-radius: 0.5rem;
  width: 20rem;
  height: 450px;
`;
export default Wrapper;
