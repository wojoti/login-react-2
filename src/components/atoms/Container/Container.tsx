import Wrapper from "./Container.style";

interface Props {
  children?: React.ReactNode;
}

function Container({ children }: Props) {
  return <Wrapper>{children}</Wrapper>;
}

export default Container;
