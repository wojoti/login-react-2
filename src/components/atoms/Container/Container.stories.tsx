import { ComponentMeta, ComponentStory } from "@storybook/react";

import Container from "./Container";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/Container",
  component: Container,
} as ComponentMeta<typeof Container>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Container> = (args) => (
  <Container {...args} />
);

export const Default = Template.bind({});
Default.args = {};
