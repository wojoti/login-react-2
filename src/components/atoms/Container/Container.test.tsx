import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Container, { ContainerProps } from "./Container";

const props: ContainerProps = {
  testId: "test-container-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<Container {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render container", () => {
  render(<Container {...props} />);
  const containerElement = screen.getByTestId("test-container-id");
  expect(containerElement).toBeInTheDocument();
});
