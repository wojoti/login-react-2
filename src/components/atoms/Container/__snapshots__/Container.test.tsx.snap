// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 0 25px 50px -12px rgb(0 0 0 / 0.25);
  padding-top: 2rem;
  padding-bottom: 2rem;
  padding-left: 2.5rem;
  padding-right: 2.5rem;
  background-color: rgb(255 255 255);
  border-style: solid;
  border-width: 0rem;
  border-radius: 0.5rem;
  width: 20rem;
  height: 450px;
}

<div
  className="c0"
  data-testid="test-container-id"
/>
`;
